### PR TITLE
Add property hostgroup_members in Hostgroup class (same for servicegr…

### LINKED ISSUE #110: Add property hostgroup_members in Hostgroup class
When comment modules (pylint), I have seen the function get_hostgroup_members of Hostgroup never works because property 'hostgroup_members' not exist.

So clean it and remove the call in function 'get_hosts_by_explosion' in same class

### PR DESCRIPTION
**Pull request by ddurieux**
…oup and contactgroup). closes #110

### CODE REVIEW

File: alignak/objects/hostgroup.py [hunk 1]
```diff
@@ -112,18 +114,13 @@ def get_hosts(self):
...
         :rtype: list
         """
-        # TODO: consistency : a Hostgroup instance should always
-        # have its hostgroup_members attribute defined, even if the empty list
         if hasattr(self, 'hostgroup_members'):
-            # consistency: any Hostgroup instance's hostgroup_members attribute
-            # should already be decoded/parsed:
-            # this should already be in its list form.
-            return [m.strip() for m in self.hostgroup_members.split(',')]
+            return self.hostgroup_members
```
Comment (Seb-Solon) [inline alignak/objects/hostgroup.py]: Do you ensure members are strip at parsing?

File: alignak/objects/contactgroup.py [hunk 2]
```diff
@@ -73,6 +73,8 @@ class Contactgroup(Itemgroup):
         'uuid':                 StringProp(default='', fill_brok=['full_status']),
         'contactgroup_name':    StringProp(fill_brok=['full_status']),
         'alias':                StringProp(fill_brok=['full_status']),
+        'contactgroup_members': ListProp(default=[], fill_brok=['full_status'],
```
Comment (Seb-Solon) [inline alignak/objects/contactgroup.py]: we should avoid this. Unless you are 100% sure this value is replaced or not use to append something. 
For example add_string_member function dos not ensure that (hasattr will return true) but add_string_unknown_member have this (bool([]) is false) 

See #403

File: alignak/objects/hostgroup.py [hunk 3]
```diff
@@ -71,14 +71,16 @@ class Hostgroup(Itemgroup):
...
-        'alias':          StringProp(fill_brok=['full_status']),
-        'notes':          StringProp(default='', fill_brok=['full_status']),
-        'notes_url':      StringProp(default='', fill_brok=['full_status']),
-        'action_url':     StringProp(default='', fill_brok=['full_status']),
-        'realm':          StringProp(default='', fill_brok=['full_status'],
-                                     conf_send_preparation=get_obj_name),
+        'uuid':                 StringProp(default='', fill_brok=['full_status']),
+        'hostgroup_name':       StringProp(fill_brok=['full_status']),
+        'alias':                StringProp(fill_brok=['full_status']),
+        'hostgroup_members':    ListProp(default=[], fill_brok=['full_status'],
```
Comment (Seb-Solon) [inline alignak/objects/hostgroup.py]: see previous comment

**Comment by mohierf** [comment]
We should merge this PR because it breaks some configuration with hostgroups  in hosts or services dependencies ...

File: alignak/objects/contactgroup.py (hunk 2 above)
Comment (mohierf) [inline alignak/objects/contactgroup.py]: I propose to leave as is ... this code is running ok actually.

Comment (mohierf) [inline alignak/objects/contactgroup.py]: There is a syntax error on the line you added ... missing parenthesis to close the ListProp!

**Comment by Seb-Solon** [comment]
"this code is running ok actually" is not a really legit excuse to merge a PR. It looks like a "it's works on my machine" answer.  Why merging if we know that it can cause issue?

**Comment by mohierf** [comment]
I do agree ... but some tests are still existing and some others are yet to come ... ddurieux needs to make some more PR for the tests that were in the former develop+enhanceunitests branch.

I understand your comment on the problems that may appear because of [] initializing but we do not match this case currently. And the bug in the groups members was a real problem in our test configuration ... and I am fed-up with making some patches to tests more important features than this one...

**Pull request merged**